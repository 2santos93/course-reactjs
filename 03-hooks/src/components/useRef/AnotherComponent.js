
import React, { forwardRef, useEffect, useImperativeHandle, useState } from 'react'

export const AnotherComponent = forwardRef((props, ref) => {
    const {inputRef, funcRef} = ref;
    const [showInput, setShowInput] = useState(true);
    useEffect(() => {
        
        inputRef.current.value = showInput ? 'encendio' : 'apagau';
        
    }, [showInput])
    
    const nelson = () => {
        console.log('me ejecute');
    }
    
    useImperativeHandle(funcRef, () => {
        return nelson;
    });

    return (
        <button 
            className='btn btn-primary'
            onClick={() => setShowInput(!showInput)}
        >
            Delete
        </button>  
    )
});
