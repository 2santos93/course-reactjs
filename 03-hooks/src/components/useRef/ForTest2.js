import React from 'react'

export const ForTest2 = () => {
  return (
    <h1>
      Im a text
    </h1>
  )
}
