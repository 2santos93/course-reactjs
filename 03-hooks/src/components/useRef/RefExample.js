import React, { useRef, useEffect } from "react";
import { AnotherComponent } from "./AnotherComponent";

export const RefExample = () => {
  const inputRef = useRef();
  const funcRef = useRef();
  useEffect(() => {
    funcRef.current();
  });

  return (
    <>
      <input type="text" ref={inputRef} />
      <AnotherComponent ref={{ inputRef, funcRef }} />
    </>
  );
};
