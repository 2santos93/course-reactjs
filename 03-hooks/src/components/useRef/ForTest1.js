import { useState } from 'react';
import { ForTest2 } from './ForTest2';

export const ForTest1 = () => {
  const [show, setShow] = useState(true);
  return (
    <>
      <button
        className="btn btn-primary"
        onClick={() => {setShow(!show)}}
      >
        {show ? 'Hide' : 'Show'}
      </button>

      {
        show && <ForTest2 />
      }
    </>
  )
}
