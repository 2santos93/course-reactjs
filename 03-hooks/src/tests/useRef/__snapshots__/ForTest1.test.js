import { RefExample } from "../../../components/useRef/RefExample";
import {shallow} from 'enzyme';
import { ForTest1 } from "../../../components/useRef/ForTest1";

describe('RefExample test', () => {
  const wrapper = shallow(<ForTest1 />);
  
  test('should render ok', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('ForTest2').exists()).toBe(true);
  });

  test('should show text', () => {
    wrapper.find('button').simulate('click');

    expect(wrapper.find('ForTest2').exists()).toBe(false);
  });
  
});
