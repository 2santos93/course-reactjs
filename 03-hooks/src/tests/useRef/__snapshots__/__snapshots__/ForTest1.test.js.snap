// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefExample test should render ok 1`] = `
<Fragment>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Hide
  </button>
  <ForTest2 />
</Fragment>
`;
